### PR TITLE
gh-89974: Support building with VS 2022

### PR DESCRIPTION
**Pull request by thatbirdguythatuknownot**
https://bugs.python.org/issue45816

* Issue: gh-89974

### CODE REVIEW

**Comment by kumaraditya303** [comment]
cc @zooba I verified this, with this python can be compiled on VS 2022 without retargeting solution.

**Comment by thatbirdguythatuknownot** [comment]
> cc @zooba I verified this, with this python can be compiled on VS 2022 without retargeting solution.

How was it verified? In this case, MSBuild v143 *must* be downloaded and installed without any of the v142 components installed with it, or else it would naturally work because v142 is available.

**Comment by kumaraditya303** [comment]
I only have v143 installed and it worked without any changes hence verified.

**Comment by thatbirdguythatuknownot** [comment]
> I only have v143 installed and it worked without any changes hence verified.

@kumaraditya303 Alright, I'm verifying that; hold on.

**Comment by thatbirdguythatuknownot** [comment]
@kumaraditya303 No it does not work:

> Using py -3.10 (found 3.10 with py.exe)
Fetching external libraries...
bzip2-1.0.6 already exists, skipping.
sqlite-3.36.0.0 already exists, skipping.
xz-5.2.2 already exists, skipping.
zlib-1.2.11 already exists, skipping.
Fetching external binaries...
libffi-3.4.2 already exists, skipping.
openssl-bin-1.1.1l already exists, skipping.
tcltk-8.6.12.0 already exists, skipping.
Finished.
Cannot find Git on PATH
Using "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\Bin\MSBuild.exe"  (found in the PATH)
Using py -3.10 (found 3.10 with py.exe)
>
> C:\Windows\system32>"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\Bin\MSBuild.exe"  "C:\Users\server\Desktop\cpython-main\PCbuild\pcbuild.proj" /t:Build /m /nologo /v:m /clp:summary /p:Configuration=Release /p:Platform=x64 /p:IncludeExternals=true /p:IncludeCTypes=true /p:IncludeSSL=true /p:IncludeTkinter=true /p:UseTestMarker=
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(436
,5): error MSB8020: The build tools for Visual Studio 2019 (Platform Toolset = 'v142') cannot be found. To build using
the v142 build tools, please install Visual Studio 2019 build tools.  Alternatively, you may upgrade to the current Vis
ual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:
\Users\server\Desktop\cpython-main\PCbuild\_freeze_module.vcxproj]

**Comment by kumaraditya303** [comment]
Apparently it was working for me locally, the windows team would verify it anyway, I am not a Visual Studio expert, I use vscode and was compiling without errors, anyways Thanks

File: Tools/msi/bundle/bootstrap/pythonba.vcxproj [hunk 1]
```diff
@@ -50,7 +51,8 @@
     </ClCompile>
     <Link>
       <AdditionalDependencies>comctl32.lib;gdiplus.lib;msimg32.lib;shlwapi.lib;wininet.lib;dutil.lib;balutil.lib;version.lib;uxtheme.lib;%(AdditionalDependencies)</AdditionalDependencies>
-      <AdditionalLibraryDirectories Condition="'$(PlatformToolset)' == 'v142'">$(WixInstallPath)sdk\vs2017\lib\x86</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories Condition="'$(PlatformToolset)' == 'v143'">$(WixInstallPath)sdk\vs2022\lib\x64</AdditionalLibraryDirectories>
```
Comment (zooba) [inline Tools/msi/bundle/bootstrap/pythonba.vcxproj]: Does this require us to bring in an updated Wix toolset? Have they made a release that include a vs2022 SDK?

Comment (thatbirdguythatuknownot) [reply Tools/msi/bundle/bootstrap/pythonba.vcxproj]: Just saw that and thought to add something. Sorry.

**Comment by zooba** [comment]
We also need the change I mentioned in this post: https://bugs.python.org/issue45816#msg406635

Otherwise official release builds will break, and it'll cause annoying delays and nasty churn in the git history when we discover it too late ;)

**Comment by thatbirdguythatuknownot** [comment]
> We also need the change I mentioned in this post: https://bugs.python.org/issue45816#msg406635
> 
> Otherwise official release builds will break, and it'll cause annoying delays and nasty churn in the git history when we discover it too late ;)

@zooba Added as part of the bug report.

**Pull request merged**